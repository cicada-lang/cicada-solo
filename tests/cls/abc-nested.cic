the_nat_type(_: Trivial): Type { Nat }

class ABC { a: Trivial, b: the_nat_type(a), c: String }

abc: ABC = {
  a: sole, b: 2, c: "c"
}

xyz: class {
  x: String,
  y: String,
  z: ABC
} = {
  x: "x",
  y: "y",
  z: abc,
}

xyz

xyz.z
xyz.z.a
xyz.z.b
xyz.z.c
