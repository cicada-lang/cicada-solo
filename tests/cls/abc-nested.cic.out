{
  x: "x"
  y: "y"
  z: {
    a: sole
    b: 2
    c: "c"
  }
}: class {
  x: String
  y: String
  z: class {
    a: Trivial
    b: Nat
    c: String
  }
}
{
  a: sole
  b: 2
  c: "c"
}: class {
  a: Trivial
  b: Nat
  c: String
}
sole: Trivial
2: Nat
"c": String
